### PR TITLE
Fixes and testing for models supported by uvb5.py

### PR DESCRIPTION
**Pull request by KC9HI**

### CODE REVIEW

File: chirp/drivers/uvb5.py [hunk 1]
```diff
@@ -674,6 +674,10 @@ def apply_limit(setting, obj):
         rs.set_apply_callback(apply_limit, self._memobj.limits)
         basic.append(rs)
 
+        # FIXME: FM Presets are currently broken
+        #        - Saved frequencies display as integers
+        #        - Can't key in a decimal point
+        #        - Even vallid entries (65, 66, 67, etc.) are not accepted
```
Comment (kk7ds) [inline chirp/drivers/uvb5.py:680]: I just merged the Float properties fix. Does that resolve this comment or is there more?

File: chirp/drivers/uvb5.py [hunk 2]
```diff
@@ -685,7 +689,7 @@ def apply_limit(setting, obj):
                 used = False
                 preset = 65
             rs = RadioSetting("fm_presets_%1i" % i, "FM Preset %i" % (i + 1),
-                              RadioSettingValueBoolean(used),
+                              # RadioSettingValueBoolean(used),
```
Comment (kk7ds) [inline chirp/drivers/uvb5.py:692]: What's this about? Is there something up with the boolean editor too?

**Review by kk7ds — COMMENTED** [review]
Did you mean to augment the testing matrix for this too? I don't think we have this model in there...

**Comment by KC9HI** [comment]
On Fri, Nov 18, 2022 at 8:50 PM Dan Smith ***@***.***> wrote:
>
> @kk7ds commented on this pull request.
>
> Did you mean to augment the testing matrix for this too? I don't think we have this model in there...

I do them separately at the end of a group now. Trying to do them
separately with each driver is what go me into trouble.

I'm trying to push it right now, but it is being [rejected]
>
> ________________________________
>
> In chirp/drivers/uvb5.py:
>
> > @@ -674,6 +674,10 @@ def apply_limit(setting, obj):
>          rs.set_apply_callback(apply_limit, self._memobj.limits)
>          basic.append(rs)
>
> +        # FIXME: FM Presets are currently broken
> +        #        - Saved frequencies display as integers
> +        #        - Can't key in a decimal point
> +        #        - Even vallid entries (65, 66, 67, etc.) are not accepted
>
> I just merged the Float properties fix. Does that resolve this comment or is there more?

Probably. If it does, I will update it again. I was just concentrating
on getting the download/upload working.

>
> ________________________________
>
> In chirp/drivers/uvb5.py:
>
> > @@ -685,7 +689,7 @@ def apply_limit(setting, obj):
>                  used = False
>                  preset = 65
>              rs = RadioSetting("fm_presets_%1i" % i, "FM Preset %i" % (i + 1),
> -                              RadioSettingValueBoolean(used),
> +                              # RadioSettingValueBoolean(used),
>
> What's this about? Is there something up with the boolean editor too?

Something else that needs to get solved or removed. It gets an error
something lise "duplicate fm_presets_0" or something like that.

Jim

**Comment by KC9HI** [comment]
> >
> > In chirp/drivers/uvb5.py:
> >
> > > @@ -685,7 +689,7 @@ def apply_limit(setting, obj):
> >                  used = False
> >                  preset = 65
> >              rs = RadioSetting("fm_presets_%1i" % i, "FM Preset %i" % (i + 1),
> > -                              RadioSettingValueBoolean(used),
> > +                              # RadioSettingValueBoolean(used),
> >
> > What's this about? Is there something up with the boolean editor too?
>
> Something else that needs to get solved or removed. It gets an error
> something lise "duplicate fm_presets_0" or something like that.
>
> Jim

This is the error...

***@***.***:~/mychirpgit/chirp$ ./chirpwx.py
Failed to import /home/jim/mychirpgit/chirp/chirp/drivers/th_uv3r:
multiple exception types must be parenthesized (wouxun.py, line 277)
Failed to import /home/jim/mychirpgit/chirp/chirp/drivers/rfinder:
multiple exception types must be parenthesized (rfinder.py, line 239)
Failed to import /home/jim/mychirpgit/chirp/chirp/drivers/bj9900:
multiple exception types must be parenthesized (bj9900.py, line 181)
Failed to import
/home/jim/mychirpgit/chirp/chirp/drivers/puxing_px888k: can only
concatenate list (not "range") to list
Failed to import /home/jim/mychirpgit/chirp/chirp/drivers/wouxun:
multiple exception types must be parenthesized (wouxun.py, line 277)
Failed to import /home/jim/mychirpgit/chirp/chirp/drivers/puxing:
multiple exception types must be parenthesized (puxing.py, line 50)
Failed to import /home/jim/mychirpgit/chirp/chirp/drivers/th_uv3r25:
multiple exception types must be parenthesized (wouxun.py, line 277)
ERROR: Radio class CSVRadio failed during detection: 'str' object has
no attribute 'decode'
ERROR: Radio class CommanderCSVRadio failed during detection:
startswith first arg must be str or a tuple of str, not bytes
ERROR: Radio class RTCSVRadio failed during detection: 'str' object
has no attribute 'decode'
ERROR: Context raised unexpected_exception
Traceback (most recent call last):
  File "/home/jim/mychirpgit/chirp/chirp/wxui/settingsedit.py", line
34, in _initialize
    self._load_settings()
  File "/home/jim/mychirpgit/chirp/chirp/wxui/settingsedit.py", line
45, in _load_settings
    self._add_group(group)
  File "/home/jim/mychirpgit/chirp/chirp/wxui/settingsedit.py", line
49, in _add_group
    propgrid = common.ChirpSettingGrid(group, self._group_control)
  File "/home/jim/mychirpgit/chirp/chirp/wxui/common.py", line 243, in __init__
    self.pg.Append(editor)
wx._core.wxAssertionError: C++ assertion "Assert failure" failed at
../src/propgrid/propgridpagestate.cpp(1691) in PrepareToAddItem():
wxPropertyGrid item with name "fm_presets_0" already exists

(chirpwx.py:7705): Gtk-CRITICAL **: 21:13:50.022:
gtk_box_gadget_distribute: assertion 'size >= 0' failed in
GtkScrollbar

(chirpwx.py:7705): Gtk-CRITICAL **: 21:13:50.067:
gtk_box_gadget_distribute: assertion 'size >= 0' failed in
GtkScrollbar

(chirpwx.py:7705): Gtk-CRITICAL **: 21:13:50.112:
gtk_box_gadget_distribute: assertion 'size >= 0' failed in
GtkScrollbar

(chirpwx.py:7705): Gtk-CRITICAL **: 21:13:50.135:
gtk_box_gadget_distribute: assertion 'size >= 0' failed in
GtkScrollbar

(chirpwx.py:7705): Gtk-CRITICAL **: 21:13:50.153:
gtk_box_gadget_distribute: assertion 'size >= 0' failed in
GtkScrollbar

(chirpwx.py:7705): Gtk-CRITICAL **: 21:13:50.176:
gtk_box_gadget_distribute: assertion 'size >= 0' failed in
GtkScrollbar

(chirpwx.py:7705): Gtk-CRITICAL **: 21:13:50.193:
gtk_box_gadget_distribute: assertion 'size >= 0' failed in
GtkScrollbar

(chirpwx.py:7705): Gtk-CRITICAL **: 21:13:50.210:
gtk_box_gadget_distribute: assertion 'size >= 0' failed in
GtkScrollbar

(chirpwx.py:7705): Gtk-CRITICAL **: 21:13:50.227:
gtk_box_gadget_distribute: assertion 'size >= 0' failed in
GtkScrollbar

(chirpwx.py:7705): Gtk-CRITICAL **: 21:13:50.244:
gtk_box_gadget_distribute: assertion 'size >= 0' failed in
GtkScrollbar

(chirpwx.py:7705): Gtk-CRITICAL **: 21:13:50.261:
gtk_box_gadget_distribute: assertion 'size >= 0' failed in
GtkScrollbar

(chirpwx.py:7705): Gtk-CRITICAL **: 21:13:50.278:
gtk_box_gadget_distribute: assertion 'size >= 0' failed in
GtkScrollbar

(chirpwx.py:7705): Gtk-CRITICAL **: 21:13:50.295:
gtk_box_gadget_distribute: assertion 'size >= 0' failed in
GtkScrollbar

(chirpwx.py:7705): Gtk-CRITICAL **: 21:13:50.312:
gtk_box_gadget_distribute: assertion 'size >= 0' failed in
GtkScrollbar
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/wx/core.py", line 3282, in <lambda>
    lambda event: event.callable(*event.args, **event.kw) )
  File "/home/jim/mychirpgit/chirp/chirp/wxui/settingsedit.py", line
34, in _initialize
    self._load_settings()
  File "/home/jim/mychirpgit/chirp/chirp/wxui/settingsedit.py", line
45, in _load_settings
    self._add_group(group)
  File "/home/jim/mychirpgit/chirp/chirp/wxui/settingsedit.py", line
49, in _add_group
    propgrid = common.ChirpSettingGrid(group, self._group_control)
  File "/home/jim/mychirpgit/chirp/chirp/wxui/common.py", line 243, in __init__
    self.pg.Append(editor)
wx._core.wxAssertionError: C++ assertion "Assert failure" failed at
../src/propgrid/propgridpagestate.cpp(1691) in PrepareToAddItem():
wxPropertyGrid item with name "fm_presets_0" already exists

**Comment by kk7ds** [comment]
Okay, lemme try to figure that out. Probably better to fix that.

**Comment by KC9HI** [comment]
On Fri, Nov 18, 2022 at 8:59 PM Jim Unroe ***@***.***> wrote:
>
> On Fri, Nov 18, 2022 at 8:50 PM Dan Smith ***@***.***> wrote:
> >
> > @kk7ds commented on this pull request.
> >
> > Did you mean to augment the testing matrix for this too? I don't think we have this model in there...
>
> I do them separately at the end of a group now. Trying to do them
> separately with each driver is what go me into trouble.
>
> I'm trying to push it right now, but it is being [rejected]

This is the error I get when trying to push my updated testing matrix

***@***.***:~/mychirpgit/chirp$ git push mychirp update-matrix
To https://github.com/KC9HI/chirp.git
 ! [rejected]          update-matrix -> update-matrix (non-fast-forward)
error: failed to push some refs to 'https://github.com/KC9HI/chirp.git'
hint: Updates were rejected because a pushed branch tip is behind its remote
hint: counterpart. Check out this branch and integrate the remote changes
hint: (e.g. 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
***@***.***:~/mychirpgit/chirp$

**Pull request merged**